src/Sarif/IFileSystem.cs

```diff
@@ -37,6 +39,17 @@ public interface IFileSystem
         string GetFullPath(string path);
 
         /// <summary>
+        /// Opens a binary file, reads all contents into a byte array, and then closes the file.
+        /// </summary>
+        /// <param name="path">
+        /// The file to open for reading.
+        /// </param>
+        /// <returns>
+        /// A byte array containing the contents of the file
+        /// </returns>
+        byte[] ReadAllBytes(string path);
```

Comment:
What nonsense. We apply this to the interface. Then we apply it to the default concrete implementation. It's already available for direct calls to the actual .NET API. None of these API are particularly abstruse in any case. 

Just a bunch of busy work that consumes vertical space. Bleagh.

Reply:
MMMV

---
In reply to: [136618018](https://github.com/Microsoft/sarif-sdk/pull/614#discussion_r136618018) [](ancestors = 136618018)

Reply:
Is this in reference to the comments or the addition of the new method signature?  

---
In reply to: [136618018](https://github.com/Microsoft/sarif-sdk/pull/614#discussion_r136618018) [](ancestors = 136618018)

Reply:
The comments.

---
In reply to: [136618958](https://github.com/Microsoft/sarif-sdk/pull/614#discussion_r136618958) [](ancestors = 136618958,136618018)

Reply:
we have a true compromise. no one is happy. everyone has agreed to drop doc comments against internal/private items. i have agreed to maintain/provide them for public.


---
In reply to: [136619704](https://github.com/Microsoft/sarif-sdk/pull/614#discussion_r136619704) [](ancestors = 136619704,136618958,136618018)